components/post_view/post_list/index.jsx

```diff
@@ -0,0 +1,448 @@
+import CreateChannelIntroMessage from 'components/post_view/channel_intro_message';
+import Constants from 'utils/constants.jsx';
+import {getNewestPostIdFromPosts, getOldestPostIdFromPosts} from 'utils/post_utils';
+
+import PostList from './post_list';
+
+const POSTS_PER_PAGE = Constants.POST_CHUNK_SIZE / 2;
+
+export default class PostListWrapper extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Array of posts in the channel, ordered from newest to oldest
+         */
+        posts: PropTypes.array,
+
+        /**
+         * The channel the posts are in
+         */
+        channel: PropTypes.object.isRequired,
+
+        /**
+         * Whether to display the channel intro at full width
+         */
+        fullWidth: PropTypes.bool,
+
+        /**
+         * Used for disabling loading more messages
+         */
+        postVisibility: PropTypes.number,
+
+        /**
+         * Used for showing more message indicators and channel intro message
+         */
+        channelPostsStatus: PropTypes.object,
+
+        /**
+         * Used for unread count
+         */
+        currentUserId: PropTypes.string,
+
+        /**
+         * Used for unread count
+         */
+        lastViewedAt: PropTypes.number,
+
+        /**
+         * Used for getting lastViewed timestamp from api.
+         */
+        member: PropTypes.object,
+
+        /**
+         * Flag used to determine if channel has to call sync actions.
+         */
+        channelSyncStatus: PropTypes.bool,
+
+        /**
+         * Used for websocket connect change to trigger sync.
+         */
+        socketStatus: PropTypes.object,
+
+        /**
+         * Used for determining if we reached to the top of channel.
+         * Used for taking a backup before clearing from redux-store.
+         */
+        postIdsInCurrentChannel: PropTypes.array,
+
+        /*
+         * Object from react-router
+         */
+        match: PropTypes.shape({
+            params: PropTypes.shape({
+                messageId: PropTypes.string,
+            }).isRequired,
+        }).isRequired,
+
+        actions: PropTypes.shape({
+
+            /*
+             * Get post for permaview
+             */
+            getPostThread: PropTypes.func.isRequired,
+
+            /*
+             * Get unreads posts onload
+             */
+            loadUnreads: PropTypes.func.isRequired,
+
+            /*
+             * Get posts using BEFORE_ID and AFTER_ID
+             */
+            loadPosts: PropTypes.func.isRequired,
+
+            /*
+             * Used for clearing posts from channel
+             */
+            clearPostsFromChannel: PropTypes.func.isRequired,
+
+            /*
+             * Used for changing posts status of channel
+             * For notifying redux store when we hit bottom or top of posts
+             */
+            channelPostsStatus: PropTypes.func.isRequired,
```

Comment:
This has the same `name` here.  Could this change starting with the action word?